pkg/ebpf/bpf_log.go

```diff
@@ -0,0 +1,194 @@
+package ebpf
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+)
+
+const BPFMaxLogFileLen = 72 // BPF_MAX_LOG_FILE_LEN
+
+type BPFLogType uint32
+
+const (
+	BPFLogIDUnspec BPFLogType = iota // BPF_LOG_ID_UNSPEC
+
+	// tracee functions
+	BPFLogIDInitContext // BPF_LOG_ID_INIT_CONTEXT
+
+	// bpf helpers functions
+	BPFLogIDMapLookupElem  // BPF_LOG_ID_MAP_LOOKUP_ELEM
+	BPFLogIDMapUpdateElem  // BPF_LOG_ID_MAP_UPDATE_ELEM
+	BPFLogIDMapDeleteElem  // BPF_LOG_ID_MAP_DELETE_ELEM
+	BPFLogIDGetCurrentComm // BPF_LOG_ID_GET_CURRENT_COMM
+	BPFLogIDTailCall       // BPF_LOG_ID_TAIL_CALL
+)
+
+var stringMap = map[BPFLogType]string{
+	BPFLogIDUnspec: "BPF_LOG_ID_UNSPEC",
+
+	// tracee functions
+	BPFLogIDInitContext: "BPF_LOG_ID_INIT_CONTEXT",
+
+	// bpf helpers functions
+	BPFLogIDMapLookupElem:  "BPF_LOG_ID_MAP_LOOKUP_ELEM",
+	BPFLogIDMapUpdateElem:  "BPF_LOG_ID_MAP_UPDATE_ELEM",
+	BPFLogIDMapDeleteElem:  "BPF_LOG_ID_MAP_DELETE_ELEM",
+	BPFLogIDGetCurrentComm: "BPF_LOG_ID_GET_CURRENT_COMM",
+	BPFLogIDTailCall:       "BPF_LOG_ID_TAIL_CALL",
+}
+
+var errorMap = map[BPFLogType]string{
+	BPFLogIDUnspec: "Unspecifed BPF log",
+
+	// tracee functions
+	BPFLogIDInitContext: "Failed to init context",
+
+	// bpf helpers functions
+	BPFLogIDMapLookupElem:  "Failed to find a map element",
+	BPFLogIDMapUpdateElem:  "Failed to add or update a map element",
+	BPFLogIDMapDeleteElem:  "Failed to delete a map element",
+	BPFLogIDGetCurrentComm: "Failed to get current command",
+	BPFLogIDTailCall:       "Failed to tail call",
+}
+
+func (b BPFLogType) String() string {
+	if bpfLog, found := stringMap[b]; found {
+		return bpfLog
+	}
+
+	return stringMap[BPFLogIDUnspec]
+}
+
+// BPFLog struct contains aggregated data about a bpf log origin
+type BPFLog struct {
+	id       BPFLogType
+	logLevel logger.Level
+	count    uint32                 // percpu log occurrences
+	padding  uint32                 // empty space
+	ret      int64                  // return value
+	cpu      uint32                 // cpu number
+	line     uint32                 // line number
+	file     [BPFMaxLogFileLen]byte // filename
+}
+
+func (b BPFLog) Error() string {
+	if bpfLog, found := errorMap[BPFLogType(b.id)]; found {
+		return bpfLog
+	}
+
+	return errorMap[BPFLogIDUnspec]
+}
+
+func (b BPFLog) ID() uint32 {
+	return uint32(b.id)
+}
+
+func (b BPFLog) Type() BPFLogType {
+	return BPFLogType(b.id)
+}
+
+func (b BPFLog) LogLevel() logger.Level {
+	return b.logLevel
+}
+
+func (b BPFLog) Count() uint32 {
+	return b.count
+}
+
+func (b BPFLog) Return() int64 {
+	return int64(b.ret)
+}
+
+func (b BPFLog) CPU() uint32 {
+	return b.cpu
+}
+
+func (b BPFLog) Line() uint32 {
+	return b.line
+}
+
+func (b BPFLog) File() []byte {
+	return b.file[:]
+}
+
+func (b BPFLog) FileAsString() string {
+	errFile := string(b.file[:])
+	errFileEnd := strings.Index(errFile, "\x00")
+
+	return errFile[:errFileEnd]
```

Comment:
nit: shouldn't we change this to logFile?

Reply:
It's leftovers. Tks. Done.